plugin/kotel/README.md

```diff
@@ -0,0 +1,151 @@
+a [`kgo.Hook`](https://pkg.go.dev/github.com/twmb/franz-go/pkg/kgo#Hook) interface. `kotel` can be used to enrich
+records with ancestor trace data for producers, or extract ancestor trace data from records for consumers. In
+addition, `kotel` tracks a variety of metrics related to connections, errors, and bytes transferred.
+
+To get started with `kotel`, you will need to set up a tracer and/or meter provider and configure the desired tracer
+and/or meter options. You can then create a `kotel` hook and pass it to the `kgo.WithHooks` options when creating a
+new client.
+
+From there, you can use the `kotel` tracing and metrics features in your `franz-go` code as needed. For more detailed
+instructions and examples, see the usage sections below.
+
+Please visit the  [OpenTelemetry documentation](https://opentelemetry.io/docs) for additional information about
+OpenTelemetry and how it can be used in your `franz-go` projects.
+
+## Tracing
+
+`kotel`'s tracing module allows you to track the lineage of `kgo.Record` objects through a series of `franz-go` hooks:
+
+1) HookProduceRecordBuffered
+2) HookProduceRecordUnbuffered
+3) HookFetchRecordBuffered
+
+To get started with tracing in `kotel`, you'll need to set up a tracer provider and configure any desired tracer
+options. You can then create a `kotel` hook and pass it to `kgo.WithHooks` when creating a new client.
+
+Here's an example of how you might do this:
+
+```go
+tracerProvider, err := initTracerProvider()
+
+tracerOpts := []kotel.TracerOpt{
+	kotel.TracerProvider(tracerProvider),
+	kotel.TracerPropagator(propagation.NewCompositeTextMapPropagator(propagation.TraceContext{})),
+}
+tracer := kotel.NewTracer(tracerOpts...)
+
+kotelOps := []kotel.Opt{
+	kotel.WithTracer(tracer)
+}
+
+kotelService := kotel.NewKotel(kotelOps...)
+
+cl, err := kgo.NewClient(
+	kgo.WithHooks(kotelService.Hooks()...),
+    // ...other opts
+)
+```
+
+`kotel` enables you to enrich records with ancestor trace data, such as trace context from a `http.Request` object that
+may already contain a parent span from an instrumentation library. This can be useful for tracking the lineage of your
+records and providing additional context for tracing and debugging purposes.
+
+To include ancestor trace data in your records, you can use the `ctx` object obtained from `tracer.Start` and pass it to
+`cl.Produce` as shown in the example below:
+
+```go
+func httpHandler(w http.ResponseWriter, r *http.Request) {
+    ctx, span := tracer.Start(r.Context(), "request-span")
+    defer span.End()
+    span.SetAttributes(attribute.String("foo", "bar"))
+
+    var wg sync.WaitGroup
+    wg.Add(1)
+    record := &kgo.Record{Topic: "my-topic", Value: []byte("bar")}
+    
+    cl.Produce(ctx, record, func(_ *kgo.Record, err error) {
+    defer wg.Done()
+    if err != nil {
+        fmt.Printf("record had a produce error: %v\n", err)
+        span.SetStatus(codes.Error, err.Error())
+        span.RecordError(err)
+    }
```

Comment:
As a code sample, this bit is formatted incorrectly -- I'd switch to tabbing for this entire function and, more importantly, add a level of indentation here.

For simplification you _could_ switch to `ProduceSync(...).First()`, but I actually prefer the wg aspect usually